Projects/Features/SearchFeature/Sources/ViewControllers/BeforeSearchContentViewController.swift

```diff
@@ -11,32 +12,52 @@ import RxSwift
 import UIKit
 import Utility
 
-protocol BeforeSearchContentViewDelegate: AnyObject {
-    func itemSelected(_ keyword: String)
-}
+// DatSource model 실제 Entity로 교체
 
-public final class BeforeSearchContentViewController: BaseStoryboardReactorViewController<BeforeSearchReactor> {
-    @IBOutlet weak var tableView: UITableView!
-    @IBOutlet weak var indicator: NVActivityIndicatorView!
+public final class BeforeSearchContentViewController: BaseReactorViewController<BeforeSearchReactor> {
+    fileprivate enum Section: Int {
+        case top
+        case mid
+        case bottom
+    }
 
-    weak var delegate: BeforeSearchContentViewDelegate?
+    enum DataSource: Hashable {
+        case youtube(model: Model)
+        case recommand(model2: Model)
+        case popularList(model: Model)
+
+        var title: String {
+            switch self {
+            case let .youtube(model):
+                return model.title
+            case let .recommand(model2):
+                return model2.title
+            case let .popularList(model):
+                return model.title
+            }
+        }
+    }
+
+    fileprivate var playlistDetailFactory: PlaylistDetailFactory!
+    fileprivate var textPopUpFactory: TextPopUpFactory!
+    fileprivate var dataSource: UICollectionViewDiffableDataSource<Section, DataSource>! = nil
 
-    var playlistDetailFactory: PlaylistDetailFactory!
-    var textPopUpFactory: TextPopUpFactory!
+    fileprivate var collectionView: UICollectionView! = nil
 
-    public static func viewController(
+    fileprivate var tableView: UITableView = UITableView().then {
```

Comment:
```suggestion
    fileprivate let tableView: UITableView = UITableView().then {
```
바뀌는게 아니면 let으로 해주면 좋겠어요

Reply:
반영했습니다.